packages/playwright-core/src/remote/playwrightConnection.ts

```diff
@@ -124,12 +126,21 @@ export class PlaywrightConnection {
   private async _initPreLaunchedBrowserMode(scope: RootDispatcher) {
     this._debugLog(`engaged pre-launched mode`);
     const playwright = this._preLaunched.playwright!;
-    const browser = this._preLaunched.browser!;
-    browser.on(Browser.Events.Disconnected, () => {
-      // Underlying browser did close for some reason - force disconnect the client.
-      this.close({ code: 1001, reason: 'Browser closed' });
-    });
-    const playwrightDispatcher = new PlaywrightDispatcher(scope, playwright, undefined, browser);
+    const browser = this._preLaunched.browser;
+    if (browser) {
+      browser.on(Browser.Events.Disconnected, () => {
+        // Underlying browser did close for some reason - force disconnect the client.
+        this.close({ code: 1001, reason: 'Browser closed' });
+      });
+    }
+    const androidDevice = this._preLaunched.androidDevice;
+    if (androidDevice) {
```

Comment:
Let's get ourselves a separate `_initPreLaunchedAndroidMode`, this browser cleanup code is not relevant here.